### ISSUE TITLE
Better ways to enable ARCore

### ISSUE THREAD

**Issue by MarcAnt01** [Magisk-Modules-Repo/ARCore_enabler#2]
At the moment to enable ARCore there doesn't seem to be a better rather than sproofing the fingerprint; this is not the best method, since some app recognize your device as a Pixel 2 and from system settings, you will see fingerprint security patches and not the ones of your phone.
Some elder versions just required some small changes in the root/data/data folder, but now they've changed the files there.
Every idea and contribution is welcome here🙂

**Comment by MarcAnt01** [comment]
> Wait, aren't you the one who works on this? Well I mean you could maybe make it so you only spoof the fingerprint to certain apps, or maybe just find the code that stops the phone from accessing the AR Core API if the device isn't supported, and just remove it.

No offense, but you look like you're talking about something you have no idea about

**Comment by Zi7ar21** [comment]
> > Wait, aren't you the one who works on this? Well I mean you could maybe make it so you only spoof the fingerprint to certain apps, or maybe just find the code that stops the phone from accessing the AR Core API if the device isn't supported, and just remove it.
> 
> No offense, but you look like you're talking about something you have no idea about

OK, so my understanding is that the fingerprint is something that an app can request…
What if you made it so that you can have it check which app is asking for it and then decide whether or not to give it the real fingerprint or the spoofed one of an AR Core supported device… The user could configure which apps get the real and which get the spoof. Is that not how a fingerprint works?

**Comment by MarcAnt01** [comment]
> > > Wait, aren't you the one who works on this? Well I mean you could maybe make it so you only spoof the fingerprint to certain apps, or maybe just find the code that stops the phone from accessing the AR Core API if the device isn't supported, and just remove it.
> > 
> > 
> > No offense, but you look like you're talking about something you have no idea about
> 
> OK, so my understanding is that the fingerprint is something that an app can request…
> What if you made it so that you can have it check which app is asking for it and then decide whether or not to give it the real fingerprint or the spoofed one of an AR Core supported device… The user could configure which apps get the real and which get the spoof. Is that not how a fingerprint works?

No, fingerprint does not work in this way and it isn't technically possible

**Comment by Zi7ar21** [comment]
> > > > Wait, aren't you the one who works on this? Well I mean you could maybe make it so you only spoof the fingerprint to certain apps, or maybe just find the code that stops the phone from accessing the AR Core API if the device isn't supported, and just remove it.
> > > 
> > > 
> > > No offense, but you look like you're talking about something you have no idea about
> > 
> > 
> > OK, so my understanding is that the fingerprint is something that an app can request…
> > What if you made it so that you can have it check which app is asking for it and then decide whether or not to give it the real fingerprint or the spoofed one of an AR Core supported device… The user could configure which apps get the real and which get the spoof. Is that not how a fingerprint works?
> 
> No, fingerprint does not work in this way and it isn't technically possible

OK then…
What part of AR Core decides to run or not?

**Comment by MarcAnt01** [comment]
Basically if apps want to check the fingerprint they do and there's no way to prevent that

**Comment by Zi7ar21** [comment]
> Basically if apps want to check the fingerprint they do and there's no way to prevent that

How do we modify them to not check or to perform a game check

**Comment by MarcAnt01** [comment]
It isn't the game, but ARCore that checks the fingerprint and whitelists it if it is among the list of supported devices

**Comment by Zi7ar21** [comment]
> It isn't the game, but ARCore that checks the fingerprint and whitelists it if it is among the list of supported devices

I edited it sorry I didn't mean game I meant fake

So, why can't we just modify Google Play Services for AR? It's open source...

**Comment by MarcAnt01** [comment]
> > It isn't the game, but ARCore that checks the fingerprint and whitelists it if it is among the list of supported devices
> 
> I edited it sorry I didn't mean game I meant fake
> 
> So, why can't we just modify Google Play Services for AR? It's open source...

It is not open source, I'd have to use reverse engineering.

**Comment by Zi7ar21** [comment]
Ah shoot I see it's just the SDK that is…
Well, how do we reverse engineer it?
https://github.com/google-ar/arcore-android-sdk

**Comment by MarcAnt01** [comment]
I have never worked with decompiled apps outside of GCam, feel free to create a patched version of ARCore and I will include it

**Comment by Zi7ar21** [comment]
> > > It isn't the game, but ARCore that checks the fingerprint and whitelists it if it is among the list of supported devices
> > 
> > 
> > I edited it sorry I didn't mean game I meant fake
> > So, why can't we just modify Google Play Services for AR? It's open source...
> 
> It is not open source, I'd have to use reverse engineering.

Oh I see, only the SDK is... https://github.com/google-ar/arcore-android-sdk

Wait I don't see code for it

So, how do we start to reverse engineer it?

**Comment by Zi7ar21** [comment]
> > > > It isn't the game, but ARCore that checks the fingerprint and whitelists it if it is among the list of supported devices
> > > 
> > > 
> > > I edited it sorry I didn't mean game I meant fake
> > > So, why can't we just modify Google Play Services for AR? It's open source...
> > 
> > 
> > It is not open source, I'd have to use reverse engineering.
> 
> Oh I see, only the SDK is... https://github.com/google-ar/arcore-android-sdk
> 
> Wait I don't see code for it
> 
> So, how do we start to reverse engineer it?

Frick the thing I sent didn't show up do I didn't see it

**Comment by MarcAnt01** [comment]
Don't use the plural, I am not interested in any overkill, at all

**Comment by Zi7ar21** [comment]
> Don't use the plural, I am not interested in any overkill, at all

?

**Comment by MarcAnt01** [comment]
> > Don't use the plural, I am not interested in any overkill, at all
> 
> ?

Reverse engineering is an overkill, if you want to try and you manage, you're welcome, but I am not going to spend my time with that.

**Comment by Zi7ar21** [comment]
Alright, I used This handy little website ([http://www.javadecompilers.com/apk](url)) which probably isn't the best method, but eh. Looking at `/resources/lib/arm64-v8a/libar_c.so` I see that different phone models have different calibration data. How does your fingerprint changing thing work? If we do modify it then do we like do we have to come up with our own calibration data?
